modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -165,6 +165,27 @@ inline void copyFromIE(const IE::Blob::Ptr &blob, MatType &mat) {
     }
 }
 
+// (taken from videoio module)
+static cv::Mutex& getInitializationMutex()
+{
+    static cv::Mutex initializationMutex;
+    return initializationMutex;
+}
+
+// (taken from DNN module)
+IE::Core& getCore(const std::string& id) {
+    static std::map<std::string, std::shared_ptr<IE::Core> > cores;
```

Comment:
why do we need multiple IE Cores? cc: @ilya-lavrenov

Reply:
In DNN, this is a workaround for HETERO plugin bug (it doesn't work well for cases with switching devices).

Global IE Core is necessary to work with MYRIAD (due to device access limitations - only exclusive access is allowed).

Reply:
@dmatveev So, if I've got it right, the redundancy of this workaround depends on whether we use HETERO plugin or not. Do we?

Reply:
@dkurt  @alalek  so we can use just a single global `IE::Core` thing right?

Reply:
@dkurt  @alalek  so we can use just a single global `IE::Core` thing right?

Reply:
> single global IE::Core

In current design, DNN module can create own IE::Core instances. So, in general, Users need to choose one between inference from G-API module or DNN module to avoid issues.

---

You can follow InferenceEngine's developers guide if it pass your tests.
And re-apply necessary workarounds from DNN later after broader testing =)

Reply:
So, I've chosen to have a single Core object as in the guide. If there are any issues in the future on MYRIAD or HETERO, I'll return a map of Cores back